src/Doctrine/Orm/Filter/SearchFilter.php

```diff
@@ -216,13 +217,44 @@ protected function filterProperty(string $property, $value, QueryBuilder $queryB
             return;
         }
 
-        $values = array_map($this->getIdFromValue(...), $values);
-
+        // association, let's fetch the entity (or reference to it) if we can so we can make sure we get its orm id
         $associationResourceClass = $metadata->getAssociationTargetClass($field);
-        $associationFieldIdentifier = $metadata->getIdentifierFieldNames()[0];
+        $associationMetadata = $this->getClassMetadata($associationResourceClass);
+        $associationFieldIdentifier = $associationMetadata->getIdentifierFieldNames()[0];
         $doctrineTypeField = $this->getDoctrineFieldType($associationFieldIdentifier, $associationResourceClass);
 
-        if (!$this->hasValidValues($values, $doctrineTypeField)) {
+        $values = array_map(function ($value) use ($associationFieldIdentifier, $doctrineTypeField) {
+            if (is_numeric($value)) {
+                return $value;
+            }
+            try {
+                $item = $this->getIriConverter()->getResourceFromIri($value, ['fetch_data' => false]);
+
+                return $this->propertyAccessor->getValue($item, $associationFieldIdentifier);
+            } catch (InvalidArgumentException) {
+                /*
```

Comment:
I actually have another version that tries to do that, but i'll leave it for another PR since it needs quite a lot more to work.

Reply:
don't worry let's go for this I think it's just find and maybe we can do better when working on #2400 next year!